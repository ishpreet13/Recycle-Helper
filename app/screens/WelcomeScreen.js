import React from "react";
import { Image, ImageBackground, StyleSheet, Text, View } from "react-native";
import AppButton from "../components/AppButton";
import Screen from "../components/Screen";

import colors from "../config/colors";

function WelcomeScreen({ navigation }) {
  // const navigation = useNavigation();
  return (
    <View style={{ flex: 1, height: "100%" }}>
      <ImageBackground
        blurRadius={0.72}
        style={styles.background}
        source={require("../assets/recycleWelcomeScreen3.jpg")}
      >
        <View style={styles.logoContainer}>
          <Image
            style={styles.logo}
            source={require("../assets/logo.png")}
          ></Image>
          <Text style={styles.tagLine}>Recycle Helper</Text>
        </View>
        <View style={styles.buttonsContainer}>
          <AppButton
            title="Login"
            onPress={() => navigation.navigate("Login")}
            color="secondary"
          ></AppButton>
          <AppButton
            title="Sign Up"
            onPress={() => navigation.navigate("SignUp")}
            color="secondary"
          ></AppButton>
        </View>
      </ImageBackground>
    </View>
  );
}

const styles = StyleSheet.create({
  background: {
    flex: 1,
    justifyContent: "flex-end",
    alignItems: "center",
  },
  buttonsContainer: {
    padding: 20,
    width: "100%",
  },
  logo: {
    width: 150,
    height: 150,
  },
  logoContainer: {
    position: "absolute",
    top: 70,
    alignItems: "center",
  },
  tagLine: {
    color: colors.white,
    fontSize: 25,
    fontWeight: "600",
    paddingVertical: 20,
  },
});
export default WelcomeScreen;
