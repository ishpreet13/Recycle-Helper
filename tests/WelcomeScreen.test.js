import "react-native";
import React from "react";
import renderer, { act } from "react-test-renderer";
import { render, fireEvent, waitFor } from "@testing-library/react-native";

import WelcomeScreen from "../app/screens/WelcomeScreen";

act(() => {
  describe("Testing WelcomeScreen ", () => {
    let wrapper;
    it("should render", async () => {
      act(() => {
        wrapper = renderer.create(<WelcomeScreen />);
      });
      expect(wrapper.toJSON()).toBeTruthy();
    });
  });
});

act(() => {
  it("WelcomeScreen should render correctly", async () => {
    let tree;
    act(() => {
      tree = renderer.create(<WelcomeScreen />);
    });
    expect(tree.toJSON()).toMatchSnapshot();
  });
});

describe("WelcomeScreen ", () => {
  it("navigates on button press", () => {
    const navigate = jest.fn();
    const { getByText } = render(<WelcomeScreen navigation={{ navigate }} />);
    fireEvent.press(getByText("Login"));
    expect(navigate).toHaveBeenCalledWith("Login");
  });
});

describe("WelcomeScreen ", () => {
  it("navigates on button press", () => {
    const navigate = jest.fn();
    const { getByText } = render(<WelcomeScreen navigation={{ navigate }} />);
    fireEvent.press(getByText("Sign Up"));
    expect(navigate).toHaveBeenCalledWith("SignUp");
  });
});
