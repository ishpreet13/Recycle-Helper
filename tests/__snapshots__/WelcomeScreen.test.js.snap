// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WelcomeScreen should render correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "height": "100%",
    }
  }
>
  <View
    accessibilityIgnoresInvertColors={true}
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "flex-end",
      }
    }
  >
    <Image
      blurRadius={0.72}
      source={1}
      style={
        Array [
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          Object {
            "height": undefined,
            "width": undefined,
          },
          undefined,
        ]
      }
    />
    <View
      style={
        Object {
          "alignItems": "center",
          "position": "absolute",
          "top": 70,
        }
      }
    >
      <Image
        source={1}
        style={
          Object {
            "height": 150,
            "width": 150,
          }
        }
      />
      <Text
        style={
          Object {
            "color": "#fff",
            "fontSize": 25,
            "fontWeight": "600",
            "paddingVertical": 20,
          }
        }
      >
        Recycle Helper
      </Text>
    </View>
    <View
      style={
        Object {
          "padding": 20,
          "width": "100%",
        }
      }
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "lightblue",
            "borderColor": "#fff",
            "borderRadius": 25,
            "justifyContent": "center",
            "marginVertical": 10,
            "opacity": 1,
            "padding": 15,
            "width": "100%",
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#fff",
                "fontSize": 20,
                "fontWeight": "bold",
                "textAlign": "center",
                "textTransform": "uppercase",
              },
              Object {
                "fontSize": 20,
              },
            ]
          }
        >
          Login
        </Text>
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "lightblue",
            "borderColor": "#fff",
            "borderRadius": 25,
            "justifyContent": "center",
            "marginVertical": 10,
            "opacity": 1,
            "padding": 15,
            "width": "100%",
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#fff",
                "fontSize": 20,
                "fontWeight": "bold",
                "textAlign": "center",
                "textTransform": "uppercase",
              },
              Object {
                "fontSize": 20,
              },
            ]
          }
        >
          Sign Up
        </Text>
      </View>
    </View>
  </View>
</View>
`;
